share/picongpu/tests/PusherScaling/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,285 @@
+
+        for iteration in indices:
+            i = self.series.iterations[iteration]  # current position
+
+            # node coordinates
+            pos2_x = i.particles["e"]["position"]["x"].load_chunk()
+            pos2_y = i.particles["e"]["position"]["y"].load_chunk()
+
+            # InCell coordinates
+            pos_off2_x = i.particles["e"]["positionOffset"]["x"].load_chunk()
+            pos_off2_y = i.particles["e"]["positionOffset"]["y"].load_chunk()
+
+            p_x = i.particles["e"]["momentum"]["x"].load_chunk()
+            p_y = i.particles["e"]["momentum"]["y"].load_chunk()
+
+            self.series.flush()
+
+            array_index = int(iteration / period)
+            # save the read data in arrays for later usage
+            x_poss[array_index] = pos2_x
+            y_poss[array_index] = pos2_y
+
+            x_offSet[array_index] = pos_off2_x
+            y_offSet[array_index] = pos_off2_y
+
+            x_momentum[array_index] = p_x
+            y_momentum[array_index] = p_y
+
+        # all radii in the series
+        radius = (
+            np.sqrt(x_momentum**2 + y_momentum**2)
+            * self.params["unit_mass"]
+            * self.params["unit_speed"]
+            / (abs(self.charge[0]) * self.params["unit_charge"] * self.B)
+            / self.params["cell_depth"]
+            / self.params["unit_length"]
+        )
+
+        R_c = radius[0]  # original radius
+
+        # half_step
+        steps_per_turn = 0.825e-12 / (self.params["dt"] * self.params["unit_time"])
+        halfStepPhase = np.pi / steps_per_turn
+
+        # coordinate transformation to reposition the origin at the center of
+        # the circle
+        x_poss = x_poss + (R_c * np.sin(halfStepPhase))
+        y_poss = y_poss - R_c + int(R_c) + (R_c - R_c * np.cos(halfStepPhase))
+
+        x_offSet = x_offSet - 32
+        y_offSet = y_offSet - 5 - int(R_c)
+
+        self.period = period
+
+        # in-cell position
+        self.x_poss = x_poss
+        self.y_poss = y_poss
+
+        # cells
+        self.x_offSet = x_offSet
+        self.y_offSet = y_offSet
+
+        # momenta of the series
+        self.x_momentum = x_momentum
+        self.y_momentum = y_momentum
+
+        self.R_c = R_c
+
+    def get_total_phasediff(self):
+        """Calculates the total phase difference accumulated during the
+        simulation.
+
+        Returns:
+        total_phasediff (float): Total phase difference
+        timestep (int): Timestep used in the series
+
+        """
+        x = self.x_poss + self.x_offSet  # x coordinates
+        y = self.y_poss + self.y_offSet  # y coordinates
+
+        res_cos = x / np.sqrt(x**2 + y**2)
+        theta = np.arccos(res_cos)
+
+        timestep = self.params["dt"] * self.params["unit_time"]
+
+        total_phasediff = 0
+        for counter in range(0, len(theta) - 1, 1):
+            total_phasediff += abs(theta[counter + 1] - theta[counter])
```

Comment:
```suggestion
            total_phasediff = np.abs(np.diff(theta))
```
to replace these three lines.